File: 卡码/代码随想录/1/06-二叉树/Js/03-二叉树修改与构造/LeetCode_0654.js
//654. 最大二叉树
import TreeNode from "../TreeNode.js";

const traversal=(nums,left,right)=>{
  if(left>=right) return null;
  let maxIndex=left;
  for(let i=left;i<right;i++){
    if(nums[i]>nums[maxIndex]) maxIndex=i;
  }
  const root=new TreeNode(nums[maxIndex]);
  root.left=traversal(nums,left,maxIndex);
  root.right=traversal(nums,maxIndex+1,right);
  return root;
};
const constructMaximumBinaryTree = function(nums) {
  return traversal(nums,0,nums.length);
};

File: 卡码/代码随想录/1/06-二叉树/C++/03-二叉树修改与构造/LeetCode_0654.cpp
//654. ×î´ó¶þ²æÊ÷
#include <iostream>
#include <vector>
#include "../TreeNode.h"
using namespace std;

class Solution {
public:
	TreeNode* traversal(vector<int>& nums,int left,int right){
		if(left>=right) return nullptr;
		int maxIndex=left;
		for(int i=left;i<right;i++)
			if(nums[i]>nums[maxIndex]) maxIndex=i;
		TreeNode* root=new TreeNode(nums[maxIndex]);
		root->left=traversal(nums,left,maxIndex);
		root->right=traversal(nums,maxIndex+1,right);
		return root;
	}
  TreeNode* constructMaximumBinaryTree(vector<int>& nums) {
		return traversal(nums,0,nums.size());
  }
};

int main(){
  system("pause");
  return 0;
}
